torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -1146,16 +1146,20 @@ class NoOpScheduler : public SchedulerEntry {
 
   //! Check if the no-op heuristics apply in given fusion
   static bool canScheduleCompileTime(Fusion* fusion) {
+    if (fusion->isNoOp()) {
+      return true;
+    }
     // Check there're no non-trivial reduction ops.
     for (auto reduction :
          ir_utils::getReductionOps(fusion, true /* ignore_trivial */)) {
-      for (auto input :
-           ir_utils::filterByType<TensorView>(reduction->inputs())) {
-        auto root_dom = input->getRootDomain();
-        auto all_nonzero =
-            std::none_of(root_dom.begin(), root_dom.end(), [](IterDomain* id) {
-              return id->extent()->isZeroInt();
-            });
```

Comment:
This was wrong. Reducing a 0 size tensor does not mean it needs no schedule. For example, if the input tensor has shape `(1000000000, 0)` and we do a `sum(dim=1)`, then we will get a tensor of shape `(1000000000,)` filled with zeros, which definitely needs parallelization.